src/passes/OptimizeInstructions.cpp

```diff
@@ -153,10 +216,13 @@ struct OptimizeInstructions
                  UnifiedExpressionVisitor<OptimizeInstructions>>> {
   bool isFunctionParallel() override { return true; }
 
-  Pass* create() override { return new OptimizeInstructions; }
+  Pass* create() override { return new OptimizeInstructions(finalize); }
 
+  bool finalize = false;
```

Comment:
as mentioned, I think we do not need the `finalize` stuff. We can consider adding it later if speed or future optimizations are an issue, I think?